onnxruntime/core/framework/cloud_invoker.cc

```diff
@@ -0,0 +1,290 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifdef USE_CLOUD
+#include "http_client.h"
+#include "core/common/common.h"
+#include "core/framework/cloud_invoker.h"
+#include "core/framework/ort_value.h"
+
+#define CHECK_TRITON_ERR(ret, msg)                                                           \
+  if (!ret.IsOk()) {                                                                         \
+    return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, msg, ", triton err: ", ret.Message().c_str()); \
+  }
+
+namespace onnxruntime {
+
+namespace tc = triton::client;
+
+constexpr char* kCloudUri = "cloud.uri";
+constexpr char* kCloudModelName = "cloud.model_name";
+constexpr char* kCloudModelVer = "cloud.model_version";
+constexpr char* kCloudVerbose = "cloud.verbose";
+constexpr char* kCloudEndpointType = "cloud.endpoint_type";
+constexpr char* kCloudAuthKey = "cloud.auth_key";
+constexpr char* kCloudTriton = "triton";
+
+class TritonInvoker : public CloudEndPointInvoker {
+ public:
+  TritonInvoker(const CloudEndPointConfig& config);
+  onnxruntime::Status Send(const CloudEndPointConfig& run_options,
+                           const InlinedVector<std::string>& input_names,
+                           gsl::span<const OrtValue> ort_inputs,
+                           const InlinedVector<std::string>& output_names,
+                           std::vector<OrtValue>& ort_outputs) const noexcept override;
+
+ private:
+  static std::string MapDataType(int32_t ort_data_type);
+  onnxruntime::TensorPtr CreateTensor(const std::string& data_type, const onnxruntime::VectorInt64& dim) const;
+
+  std::string uri_;
+  std::string model_name_;
+  std::string model_ver_ = "1";
+  bool verbose_ = false;
+  std::shared_ptr<CPUAllocator> cpu_allocator_;
```

Comment:
Why do we need a separate allocator here? The CPU allocator is arena based and will consume a lot of memory.

Reply:
Now used passed-in from session-state.